modules/build/src/main/scala/scala/build/preprocessing/directives/UsingCustomJarDirectiveHandler.scala

```diff
@@ -38,20 +39,27 @@ case object UsingCustomJarDirectiveHandler extends UsingDirectiveHandler {
 
   override def keys = Seq("jar", "jars")
   override def handleValues(
-    values: Seq[Any],
-    cwd: ScopePath,
-    positionOpt: Option[Position]
+    values: Seq[Value[_]],
+    cwd: ScopePath
   ): Either[BuildException, BuildOptions] = either {
+    val extraJars: Seq[Either[BuildException, os.Path]] = DirectiveUtil.stringValues(values).map {
+      case (p, pos) =>
+        val root = Directive.osRoot(cwd, Some(pos))
+        // FIXME Handle malformed paths here
+        root.map(os.Path(p, _))
+    }
 
-    val root = value(Directive.osRoot(cwd, positionOpt))
-    val extraJars = DirectiveUtil.stringValues(values).map { p =>
-      // FIXME Handle malformed paths here
-      os.Path(p, root)
+    val errors = extraJars.collect {
+      case Left(error) => error
     }
 
+    val res = if (errors.nonEmpty) Left(CompositeBuildException(errors)) else Right(extraJars.collect {
+      case Right(value) => value
+    })
+
```

Comment:
For some reason, I couldn't use `partitionMap`. Compiler threw an error that `partitionMap` is not a member of `Seq[Either[A, B]]`

Reply:
It was introduced after 2.12.x